src/server/server.ts

```diff
@@ -173,17 +184,62 @@ module ts.server {
             });
 
             rl.on('close',() => {
-                this.projectService.closeLog();
                 this.projectService.log("Exiting...");
+                this.projectService.closeLog();
                 process.exit(0);
             });
         }
     }
 
+    interface LogEnv {
+        file?: string;
+        level?: string;
+    }
+
+    function parseLogEnv(logEnvStr: string): LogEnv {
+        var logEnv: LogEnv = {};
+        var args = logEnvStr.split(' ');
+        for (var i = 0, len = args.length; i < len; i++) {
+            var option = args[i];
+            var value = args[i + 1];

```

Comment:
I think you mean to be incrementing `i` by `2` each time, and need to check your bounds appropriately (i.e. `i < len - 1`).